README.md

```diff
@@ -113,6 +113,27 @@ fplll can also be built from source. Below, we explicate some of the dependencie
   
 NOTE: If you are intending to use fplll on Windows 10, then these packages should be installed after the `Windows Subsystem for Linux` has been installed and activated. Please go to the [Windows 10](#windows-10) instructions for more information.
 
+### Using `cmake` instead of `libtools` and `auto[conf|make]` ###
+
+_Note:_ the `CMakeLists.txt` configuration is unofficial and trying to reproduce the official build as close as possible, however, less tested cross-platforms.
+
+One upside of `cmake` is, it will generate compilation database automatically at `build/compile_commands.json` which integrates well with language server like `clangd`.
```

Comment:
Two things, mostly for interest:

1. I don't think it's true that `cmake` generates a compile database automatically. As far as I'm aware, you need to set `CMAKE_EXPORT_COMPILE_COMMANDS`, which isn't quite automatic.
2. You can actually mimic the same thing in a compiler-independent fashion by using [bear](https://github.com/rizsotto/Bear).

Reply:
re: `bear`

what drives me to spend time migrating to cmake is frustration with bear which doesn't work well with `.lo` and `.la` object from libtools, but mostly works for `.o` object files

and yes, you do need to set flags normally (but I already added `set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`, so should be on always.